src/ucp/wireup/wireup.c

```diff
@@ -808,7 +801,25 @@ ucp_wireup_connect_lane_to_iface(ucp_ep_h ep, ucp_lane_index_t lane,
         return status;
     }
 
-    ucp_wireup_assign_lane(ep, lane, uct_ep, "");
+    /* If EP already exists, it's a wireup proxy (only client-server),
+     * and we need to update its next_ep instead of replacing it.
+     */
+    if (ep->uct_eps[lane] == NULL) {
```

Comment:
can we move the block of code in line 776..786 to here, and unite with it?
some logic is duplicated